terraform/services/alarm-to-slack/lambda_src/lambda_function.py

```diff
@@ -34,65 +85,86 @@ def cloudwatch_message(record):
 
         if message.get('OldStateValue'):
             return message
-    except JSONDecodeError:
-        log({'messageId': record.get('messageId'),
-             'msg': 'Did not receive an SNS Cloudwatch payload',})
+    except json.JSONDecodeError:
+        log({'msg': 'Did not receive an SNS Cloudwatch payload',
+             'messageId': record.get('messageId')})
     return None
 
 def enriched_cloudwatch_message(record):
     """
-    Logs the CloudWatch message (if it exists).
-    Enriches the message with an emoji for display.
+    Parses the CloudWatch message, extracts the app and env from the alarm name,
+    and enriches it with an emoji for display.
     """
     message = cloudwatch_message(record)
     if message:
-        log({'messageId': record.get('messageId'),
-             'AlarmName': message.get('AlarmName'),
-             'NewStateValue': message.get('NewStateValue'),
-             'OldStateValue': message.get('OldStateValue'),
-             'StateChangeTime': message.get('StateChangeTime'),
-             'msg': 'Received CloudWatch Alarm',
-            })
-        if message['NewStateValue'] == 'OK':
-            if os.environ.get('IGNORE_OK') == 'true':
+        alarm_name = message.get('AlarmName')
+        if alarm_name:
+            match = re.match(r'^(?P<app>[a-zA-Z0-9]+)-.*', alarm_name)
+            if match:
+                app_name = match.group('app')
+                message['App'] = app_name
+
+                env_match = re.match(
+                    r'^[a-zA-Z0-9]+-(?P<env>[a-zA-Z0-9]+)-.*',
+                    alarm_name)
+                if env_match:
+                    message['Env'] = env_match.group('env')
+
+                log({'App': app_name,
+                     'AlarmName': alarm_name,
+                     'NewStateValue': message.get('NewStateValue'),
+                     'OldStateValue': message.get('OldStateValue'),
+                     'StateChangeTime': message.get('StateChangeTime'),
+                     'msg': 'Received CloudWatch Alarm',
+                     'messageId': record.get('messageId')})
+
+                if message['NewStateValue'] == 'OK' and app_name in ignore_ok_apps:
+                    return None
+
+                if message['NewStateValue'] == 'OK':
+                    message['Emoji'] = ':checked:'
+                else:
+                    message['Emoji'] = ':anger:'
+            else:
+                log({'msg': f'AlarmName "{alarm_name}" does not match expected format',
+                     'messageId': record.get('messageId')})
                 return None
-            message['Emoji'] = ':checked:'
         else:
-            message['Emoji'] = ':anger:'
+            log({'msg': 'AlarmName not found in message',
+                 'messageId': record.get('messageId')})
+            return None
```

Comment:
Please add tests that check the "AlarmName does not match" and "AlarmName not found" paths.